### PR TITLE
JEP Draft: “Java 10 and Java 11 support in Jenkins”

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This JEP draft summarizes the current plan regarding Java 10 (and tentatively Java 11) support in Jenkins. This JEP depends on #135 which would be a venue for the JEP discussion.

@jenkinsci/jep-editors @kohsuke @bitwiseman @tracymiranda

### CODE REVIEW

File: jep/0000/README.adoc [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+| Not Submitted :information_source:
+
+| Type
+| Process
+
+| Created
+| 2018-07-05
+
+| BDFL-Delegate
+| :bulb: Link to github user page :bulb:
```
Comment (bitwiseman) [inline jep/0000/README.adoc]: Do you have any thoughts on who should be the BDFL-Delegate for this plan?

File: jep/0000/README.adoc [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+== Motivation
+
+Non-goals in the specification are defined to limit the scope of work.
+The main objective is to get Jenkins running with Java 10+,
+there will be follow-up tasks to cleanup Illegal Reflective Access warnings and to adopt new features.
+
+== Reasoning
+
+Reasoning for the Specification decisions is provided in the “Approach” section above.
```
Comment (bitwiseman) [inline jep/0000/README.adoc]: I don't see an "approach" section, but also reasoning goes here not somewhere else.

File: jep/0000/README.adoc (hunk 1 above)
Comment (oleg-nenashev) [reply jep/0000/README.adoc]: Options: 

* @kohsuke as BDFL and an initiator of the hackathon
* @daniel-beck / @olivergondza as weekly/LTS maintainers
* @dwnusbaum / @ndeloof / @svanoort as active contributors

But generally I would expect the most of the discussion to happen within the Platform SiG (#135), which could assign a BDFL delegate

File: jep/0000/README.adoc (hunk 2 above)
Comment (oleg-nenashev) [reply jep/0000/README.adoc]: I renamed it to "Goals and non-goals" in specification.
IMHO it fits better as a part of specification, not Reasoning

File: jep/0000/README.adoc [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+=== Jenkins core patches
+
+Must-have stories are defined in link:https://issues.jenkins-ci.org/browse/JENKINS-52012[JENKINS-52012].
+All stories in this EPIC need to be completed.
+
+==== Library updates
+
+* The JENKINS-52012 EPIC includes a number of library updates in the core we know about: Groovy, ASM, etc.
+* Some updates may require downstream plugin updates.
+** For Example, Groovy update requires cleanup of the Metaspace leak memory in Script Security
```
Comment (svanoort) [inline jep/0000/README.adoc]: Also in Pipeline plugins.

File: jep/211/README.adoc [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+== Abstract
+
+In September 2018 we expect Java 11 to be released.
+It will be an LTS version with a long support timeline.
+This jenkins Enhancement Proposal describes actions required in order to make Java 10 and 11 releases publicly available.
```
Comment (dwnusbaum) [inline jep/211/README.adoc]: jenkins -> Jenkins

**Pull request merged**